### ISSUE TITLE
`missing-docs` triggers for missing module docs in build scripts

### ISSUE THREAD

**Issue by wmmc88** [rust-lang/rust#124343]
I created an empty `build.rs` file and build with `-W missing-docs` and it triggered this warning:
```
warning: missing documentation for the crate
 --> crates\wdk-alloc\build.rs:4:1
  |
4 | fn main() {}
  | ^^^^^^^^^^^^
  |
  = note: requested on the command line with `-W missing-docs`
```

AFAIU, buildscripts functions/modules/etc can't show up in generated rustdocs, so should this warning not trigger?

### Meta

`rustc --version --verbose`:
```
rustc 1.77.2 (25ef9e3d8 2024-04-09)
binary: rustc
commit-hash: 25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04
commit-date: 2024-04-09
host: x86_64-pc-windows-msvc
release: 1.77.2
LLVM version: 17.0.6
```

This also happens on latest nightly.

**Comment by fmease** [comment]
I think this isn't the right place to report this, the [rust-lang/cargo issue tracker](https://github.com/rust-lang/cargo/issues/) looks like a better place for this.

`rustc` doesn't know about build scripts, it's Cargo concept.
